### PR TITLE
[VectorDistribution] Split layout configuration and distribution

### PR DESCRIPTION
**Pull request by Groverkss**
This patch splits the LLVMGPUVectorDistribution pass into two separate passes, one that sets the layouts and one that distributes. This improves the debugging experience and the failing IR can be checked for the anchors.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.td [hunk 1]
```diff
@@ -60,6 +60,12 @@ def LLVMGPUCastTypeToFitMMA : InterfacePass<"iree-llvmgpu-cast-type-to-fit-mma",
   let constructor = "mlir::iree_compiler::createLLVMGPUCastTypeToFitMMAPass()";
 }
 
+def LLVMGPUConfigureVectorLayouts :
+    InterfacePass<"iree-llvmgpu-configure-vector-layouts", "mlir::FunctionOpInterface"> {
+  let summary = "Pass to distribute vectorized functions.";
```
Comment (raikonenfnu) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.td]: NIT, I think need updated summary

File: compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorDistribute.cpp [hunk 2]
```diff
@@ -33,347 +22,25 @@
...
-                                 ArrayRef<int64_t> workgroupSize,
-                                 IREE::GPU::MMAScheduleAttr schedule,
-                                 Value laneId, int64_t subgroupSize,
-                                 bool printLayout)
-      : VectorLayoutOptions(root, /*fullConversion=*/!printLayout),
-        workgroupSize(workgroupSize), schedule(schedule),
-        printLayout(printLayout), patterns(root->getContext()) {
+  ContractionVectorLayoutOptions(Operation *root, Value laneId,
+                                 int64_t subgroupSize)
+      : VectorLayoutOptions(root), patterns(root->getContext()) {
```
Comment (raikonenfnu) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorDistribute.cpp:33]: NIT: Seems like we don't need root as part of the vectorLayoutOptions or ContractionVectorLayoutOptions anymore? Maybe we can remove it, but we can do that in following PRs.

**Review by raikonenfnu — APPROVED** [review]
Overall LGTM, just couple comments and NITs.

File: compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.td (hunk 1 above)
Comment (Groverkss) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.td]: done

File: compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorDistribute.cpp (hunk 2 above)
Comment (Groverkss) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorDistribute.cpp:33]: Yeah we probably don't need VectorLayoutOptions anymore. I need to do a cleanup, can do after this patch.

**Pull request merged**